redfish/resource_redfish_bios.go

```diff
@@ -0,0 +1,268 @@
+package redfish
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/common"
+	"github.com/stmcginnis/gofish/redfish"
+	"log"
+	"strconv"
+	"strings"
+)
+
+func resourceRedfishBios() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceRedfishBiosUpdate,
+		ReadContext:   resourceRedfishBiosRead,
+		UpdateContext: resourceRedfishBiosUpdate,
+		DeleteContext: resourceRedfishBiosDelete,
+		Schema: map[string]*schema.Schema{
+			"attributes": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "Bios attributes",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"settings_apply_time": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The time when the BIOS settings can be applied. Applicable values are 'OnReset', 'Immediate', 'AtMaintenanceWindowStart' and 'InMaintenanceWindowStart'.",
+				ValidateFunc: validation.StringInSlice([]string{
+					string(common.ImmediateApplyTime),
+					string(common.OnResetApplyTime),
+					string(common.AtMaintenanceWindowStartApplyTime),
+					string(common.InMaintenanceWindowOnResetApplyTime),
+				}, false),
+			},
+
+			"bios_config_job_uri": {
+				Type:        schema.TypeString,
+				Description: "BIOS configuration job uri",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceRedfishBiosUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	log.Printf("[DEBUG] Beginning update")
+	var diags diag.Diagnostics
+
+	conn := m.(*gofish.APIClient)
+
+	// check if there is already a bios config job in progress
+	// if yes, then check the current status of the job. If it
+	// has not completed yet, then don't perform another operation
+	pending := false
```

Comment:
Here maybe we could change a little the behavior.
If there is a job scheduled, and you call this method, silently it won't trigger any warning, which may be confusing for the user. We might trigger a warning using **diag.Diagnostics** to let the user know that no operation will be performed until that job is finished.

Reply:
@mikeletux, I agree. I will make the changes